libminifi/src/core/ContentRepository.cpp

```diff
@@ -60,16 +60,29 @@ void ContentRepository::incrementStreamCount(const minifi::ResourceClaim &stream
   }
 }
 
+void ContentRepository::removeFromPurgeList() {
+  for (auto it = purge_list_.begin(); it != purge_list_.end();) {
+    if (removeKey(*it)) {
+      purge_list_.erase(it++);
+    } else {
+      ++it;
+    }
+  }
+}
```

Comment:
This is a nice mechanism, but I have one concern: What if we just keep getting errors due to some persistent issue?
Can we limit the memory consumed by this list? E.g. stop operation if it grows above 1M or so entries.
Can we keep retrying deletion of the purge list after an agent restart or flow update?

Reply:
It depends if we expect any persistent issues that have to be dealt with. My thinking was that we are creating this list for any temporary issues that should be resolved in a few tries, and it should only affect a small number of entries. In case of a restart I think the `clearOrphans` mechanism should remove the entries we did not manage to delete before, maybe if that fails those can be added to the purge list for handling them later. What do you think? Should we discuss this with the team?

Reply:
Let's stay with no limit for now, but implement retry in `clearOrphans` as well, so an agent restart doesn't cause content leaks.

Reply:
Added purge list update in `clearOprhans` in bde565c405e3e9419020445a48d41da561b58729